### PR TITLE
Use batch for historical confirmations filters request.

### PR DESCRIPTION
**Pull request by roeierez**
This PR introduces an optimization for the filters fetching used in historical confirmations notifications.
Instead of fetching the filters one by one it now uses a batched request and limit the batch size to the total needed items.

Fixes https://github.com/lightningnetwork/lnd/issues/4060
Depends on https://github.com/lightninglabs/neutrino/pull/203

### CODE REVIEW

File: chainntnfs/neutrinonotify/neutrino.go [hunk 1]
```diff
@@ -528,6 +528,8 @@ func (n *NeutrinoNotifier) historicalConfDetails(confRequest chainntnfs.ConfRequ
 		regFilter, err := n.p2pNode.GetCFilter(
 			*blockHash, wire.GCSFilterRegular,
 			neutrino.NumRetries(5),
+			neutrino.OptimisticReverseBatch(),
+			neutrino.MaxBatchSize(scanHeight-startHeight+1),
```
Comment (wpaulino) [inline chainntnfs/neutrinonotify/neutrino.go]: Maybe note that this won't end up fetching the whole batch every iteration due to a cache being used.

Comment (roeierez) [reply chainntnfs/neutrinonotify/neutrino.go]: Done.

**Comment by wpaulino** [comment]
@roeierez the Neutrino PR has been merged now, could you also modify the commit message to fit our standard?

**Comment by roeierez** [comment]
@wpaulino sure I can do that.
I think the first commit message is missing "build: " prefix. 
Can you please point what other changes are needed?

**Comment by wpaulino** [comment]
That's all.

File: go.mod [hunk 2]
```diff
@@ -64,4 +64,6 @@ replace github.com/lightningnetwork/lnd/cert => ./cert
 
 replace git.schwanenlied.me/yawning/bsaes.git => github.com/Yawning/bsaes v0.0.0-20180720073208-c0276d75487e
 
+replace github.com/lightninglabs/neutrino => github.com/breez/neutrino v0.11.1-0.20200312075431-42a0bdf361ba
```
Comment (wpaulino) [inline go.mod]: We need to get rid of the replace statement now and point neutrino to master.

Comment (roeierez) [reply go.mod]: neutrino dependency has been updated to master.

**Comment by wpaulino** [comment]
Looks like the build failed due to a dirty git staging area, you probably need to run `go mod tidy`.

**Comment by wpaulino** [comment]
@roeierez looks good now, except the commits are out of order again.

**Review by FeliciaXmL — COMMENTED** [review]
compare

**Pull request merged**